src/main/java/com/example/demo/services/UserService.java

```diff
@@ -0,0 +1,82 @@
+package com.example.demo.services;
+
+import com.example.demo.dtos.ProductDTO;
+import com.example.demo.entities.User;
+import com.example.demo.mappers.DtoMapper;
+import com.example.demo.products.Product;
+import com.example.demo.repositories.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+@Service
+@RequiredArgsConstructor
+public class UserService {
+
+    @Autowired
+    public static UserRepository userRepository;
+    private final DtoMapper mapper;
+
+    public Page<User> pageUsers(Pageable pageRequest){
+        return userRepository.findAll(pageRequest);
+    }
+
+    public void banUser(Long id) {
+        userRepository.findById(id).ifPresent(User -> {
+            User.setActive(false);
+            userRepository.save(User);
+        });
+    }
+
+    public void addProduct(User user, Product product) {
```

Comment:
I doubt that Product-related operations need to be performed in UserService. I'm not sure that his design is good though. I'd make Product a "first-class citizen" and set userId as a field.